### PR TITLE
Add Environment Fallback Hosts

### LINKED ISSUE #87: Defaults: Generate environment fallbacks
Per `RSC15g`, introduced in https://github.com/ably/docs/pull/965.

See https://github.com/ably/ably-js/pull/682 and https://github.com/ably/ably-ruby/pull/196 for reference implementations.

┆Issue is synchronized with this [Jira Story](https://ably.atlassian.net/browse/SDK-512) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by sacOO7**
- Refactored `ClientOptions` 
- Added defaults into separate file `Defaults.php` 
- Added support for environment fallbacks
- Fixed #87

### CODE REVIEW

File: src/Models/ClientOptions.php [hunk 1]
```diff
@@ -60,7 +62,7 @@ class ClientOptions extends AuthOptions {
     /**
      * @var string[] fallback hosts, used when connection to default host fails, populated automatically
      */
-    public $fallbackHosts;
+    public $fallbackHosts = [] ;
```
Comment (owenpearson) [inline src/Models/ClientOptions.php:65]: Formatting is a bit funny here

File: tests/DefaultsTest.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            "sandbox-a-fallback.ably-realtime.com",
+            "sandbox-b-fallback.ably-realtime.com",
+            "sandbox-c-fallback.ably-realtime.com",
+            "sandbox-d-fallback.ably-realtime.com",
+            "sandbox-e-fallback.ably-realtime.com"
+        ];
+        $fallbackHosts = Defaults::getEnvironmentFallbackHosts("sandbox");
+        $this->assertEquals($expectedFallbackHosts, $fallbackHosts);
+    }
+}
```
Comment (owenpearson) [inline tests/DefaultsTest.php:37]: Missing a new line at the end of this file for some reason

Comment (owenpearson) [reply tests/DefaultsTest.php:37]: OK I just noticed this is true for all the files you created, please fix all instances of this

**Review by owenpearson — CHANGES_REQUESTED** [review]
Looks great 👍  A couple of formatting issues to fix and then we're good

File: src/Models/ClientOptions.php (hunk 1 above)
Comment (sacOO7) [reply src/Models/ClientOptions.php]: Fixed

File: tests/DefaultsTest.php (hunk 2 above)
Comment (sacOO7) [reply tests/DefaultsTest.php]: Done

**Review by owenpearson — APPROVED** [review]
LGTM, thanks

**Review by ikbalkaya — APPROVED** [review]
LGTM

**Pull request merged**